lib/domain/contact_manager/contacts_manager.dart

```diff
@@ -27,40 +55,151 @@ class ContactsManager {
   Set<Contact> get phonebookContacts => _phonebookContacts;
 
   bool get firstSynchronizing =>
-      _syncContactsState == SyncContactsState.synchronizing;
+      _syncContactsState == SyncContactsState.synchronizing ||
+      _syncContactsState == SyncContactsState.initial;
 
   bool get firstSynchronized =>
       _syncContactsState == SyncContactsState.synchronized;
 
-  StreamSubscription? contactsStreamSubscription;
+  StreamController<ContactsManagerArguments> contactsStream =
+      StreamController<ContactsManagerArguments>.broadcast();
 
-  void initialSynchronizeContacts() {
+  ContactsManager() {
+    FlutterContacts.addListener(_fetchPhonebookContacts);
+  }
+
+  void initialSynchronizeContacts() async {
+    if (PlatformInfos.isMobile && !_isNotDisplayWarningAgainCacheMemory) {
+      await _handleRequestContactsPermission();
+    }
     if (!firstSynchronized) {
       _syncContactsState = SyncContactsState.synchronizing;
       _getAllContacts();
+      _fetchPhonebookContacts();
+    } else {
+      _handleContactsHasDataChange();
+    }
+  }
+
+  Future<void> _handleRequestContactsPermission() async {
+    final currentContactsPermissionStatus =
+        await permissionHandlerService.requestContactsPermissionActions();
+    if (currentContactsPermissionStatus == PermissionStatus.granted) {
+      _isNotDisplayWarningAgainCacheMemory = true;
+      _displayWaringContactsBannerState = DisplayWaringContactsBannerState.hide;
+    } else {
+      if (!_isNotDisplayWarningAgainCacheMemory) {
+        _displayWaringContactsBannerState =
+            DisplayWaringContactsBannerState.show;
+      }
     }
   }
 
   void _getAllContacts() {
-    contactsStreamSubscription = _getAllContactsInteractor
+    _getAllContactsInteractor
         .execute(limit: AppConfig.maxFetchContacts)
         .listen((event) {
       event.fold(
         (_) {},
         (success) {
           if (success is GetContactsSuccess) {
             _tomContacts = success.tomContacts.toSet();
-            _phonebookContacts = {};
-            _syncContactsState = SyncContactsState.synchronized;
+            _handleContactsHasDataChange();
             Logs().d(
               "ContactManagerMixin()::getAllContacts(): TomContacts: ${_tomContacts.length}",
             );
-            Logs().d(
-              "ContactManagerMixin()::getAllContacts(): PhonebookContacts: ${_phonebookContacts.length}",
-            );
           }
         },
       );
     });
   }
+
+  void _fetchPhonebookContacts() async {
+    final phonebookContacts =
+        await FlutterContacts.getContacts(withProperties: true);
```

Comment:
depend the presentation to the dedicated 3rd party is not a good choice. Please add the interactor with the repository. Thank you

Reply:
done